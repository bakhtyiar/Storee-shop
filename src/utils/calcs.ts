import {ICart} from "./server-api/cart/cart.types";
import {IProduct} from "./server-api/products/products.types";

export const cartRecalcCounters = (cart: ICart): ICart => {
    cart.total = cart.products.reduce(
        (accumulator: number, product: IProduct) => (
            accumulator + (product.price * (product.quantity ?? 1))), 0)
    cart.discountedTotal = cart.products.reduce(
        (accumulator: any, product: any) => (
            accumulator + Math.round(product.price * ((100 - product.discountPercentage) / 100)) * product.quantity), 0);
    cart.totalProducts = cart.products.length;
    cart.totalQuantity = cart.products.reduce(
        (accumulator: any, product: any) => (
            accumulator + product.quantity), 0) || cart.totalProducts;
    return cart;
}