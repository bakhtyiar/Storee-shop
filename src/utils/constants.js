export const dummyjsonURL = `https://dummyjson.com`;
export const pageLimit = 30;
export const routes = {
	home: { path: "/" },
	products: { path: "/products" },
	product: { path: "/product" },
	news: { path: "/news" },
	cart: { path: "/cart"},
	orderMaking: {path: "/orderMaking"},
	orderCompleted: {path: "/order_completed"},
	post: { path: "/post" },
	profile: { path: "/profile"},
	login: {path: "/login"},
	register: {path: "/register"},
	notFound: { path: "/page_not_found" },
};
export const encryptKey = 'StoreeEncryptKey2023';
export const states = [
	{ value: 'AL', label: 'Alabama (AL)' },
	{ value: 'AK', label: 'Alaska (AK)' },
	{ value: 'AZ', label: 'Arizona (AZ)' },
	{ value: 'AR', label: 'Arkansas (AR)' },
	{ value: 'CA', label: 'California (CA)' },
	{ value: 'CO', label: 'Colorado (CO)' },
	{ value: 'CT', label: 'Connecticut (CT)' },
	{ value: 'DE', label: 'Delaware (DE)' },
	{ value: 'FL', label: 'Florida (FL)' },
	{ value: 'GA', label: 'Georgia (GA)' },
	{ value: 'HI', label: 'Hawaii (HI)' },
	{ value: 'ID', label: 'Idaho (ID)' },
	{ value: 'IL', label: 'Illinois (IL)' },
	{ value: 'IN', label: 'Indiana (IN)' },
	{ value: 'IA', label: 'Iowa (IA)' },
	{ value: 'KS', label: 'Kansas (KS)' },
	{ value: 'KY', label: 'Kentucky (KY)' },
	{ value: 'LA', label: 'Louisiana (LA)' },
	{ value: 'ME', label: 'Maine (ME)' },
	{ value: 'MD', label: 'Maryland (MD)' },
	{ value: 'MA', label: 'Massachusetts (MA)' },
	{ value: 'MI', label: 'Michigan (MI)' },
	{ value: 'MN', label: 'Minnesota (MN)' },
	{ value: 'MS', label: 'Mississippi (MS)' },
	{ value: 'MO', label: 'Missouri (MO)' },
	{ value: 'MT', label: 'Montana (MT)' },
	{ value: 'NE', label: 'Nebraska (NE)' },
	{ value: 'NV', label: 'Nevada (NV)' },
	{ value: 'NH', label: 'New Hampshire (NH)' },
	{ value: 'NJ', label: 'New Jersey (NJ)' },
	{ value: 'NM', label: 'New Mexico (NM)' },
	{ value: 'NY', label: 'New York (NY)' },
	{ value: 'NC', label: 'North Carolina (NC)' },
	{ value: 'ND', label: 'North Dakota (ND)' },
	{ value: 'OH', label: 'Ohio (OH)' },
	{ value: 'OK', label: 'Oklahoma (OK)' },
	{ value: 'OR', label: 'Oregon (OR)' },
	{ value: 'PA', label: 'Pennsylvania (PA)' },
	{ value: 'RI', label: 'Rhode Island (RI)' },
	{ value: 'SC', label: 'South Carolina (SC)' },
	{ value: 'SD', label: 'South Dakota (SD)' },
	{ value: 'TN', label: 'Tennessee (TN)' },
	{ value: 'TX', label: 'Texas (TX)' },
	{ value: 'UT', label: 'Utah (UT)' },
	{ value: 'VT', label: 'Vermont (VT)' },
	{ value: 'VA', label: 'Virginia (VA)' },
	{ value: 'WA', label: 'Washington (WA)' },
	{ value: 'WV', label: 'West Virginia (WV)' },
	{ value: 'WI', label: 'Wisconsin (WI)' },
	{ value: 'WY', label: 'Wyoming (WY)' }
]
export const shipmentMethods = {
	'warehouse': {
		value: 'warehouse',
		label: 'Pick up from warehouse',
	},
	'courier': {
		value: 'courier',
		label: 'Courier delivery',
	},
	'postOffice': {
		value: 'postOffice',
		label: 'Delivery to the post office',
	},
	'airDrone': {
		value: 'airDrone',
		label: 'Air drone shipment',
	},
}
export const warehouses = [
	{
		address: 'California Main st. 11',
		id: '001',
	},
	{
		address: 'Kansas Main st. 25',
		id: '002',
	},
	{
		address: 'Ohio Main st. 7',
		id: '003',
	},
	{
		address: 'Washington Main st. 2',
		id: '004',
	},
	{
		address: 'Louisiana Main st. 21',
		id: '005',
	},
	{
		address: 'Hawaii Main st. 6',
		id: '006',
	},
]
export const paymentMethods = {
    'card': {
		value: 'card',
		label: 'Debit card',
	},
	'cash': {
		value: 'cash',
		label: 'Cash on receive',
	},
	'inShare': {
		value: 'inShare',
		label: 'Payment in shares',
	},
}