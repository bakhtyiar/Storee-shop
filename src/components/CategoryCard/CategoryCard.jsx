import React from 'react';
import {routes} from "../../utils/constants";
import {useNavigate} from "react-router-dom";
import {Card} from "react-bootstrap";
import {capitalizeStr, replaceDashToSpace} from "../../utils/str/str";

const CategoryCard = (props) => {
    let header = props.header;
    let navigate = useNavigate();
    header = capitalizeStr(header);
    header = replaceDashToSpace(header);

    return (
        <div
            onClick={() => navigate(`${routes.products.path}/1?category=${props.header}`)}
        >
            <Card
                bg={'light'}
                className="mb-2"
            >
                <Card.Body>
                    <Card.Title className='m-0'>{header}</Card.Title>
                </Card.Body>
            </Card>
        </div>
    );
};

export default CategoryCard;
