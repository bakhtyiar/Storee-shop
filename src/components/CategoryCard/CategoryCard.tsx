import React from 'react';
import {routes} from "../../utils/constants";
import {useNavigate} from "react-router-dom";
import {Card} from "react-bootstrap";
import {capitalizeStr, replaceDashToSpace} from "../../utils/str/str";
import styled from "styled-components";

const StyledCard = styled(Card)`
  cursor: pointer;
  transition: all ease-in-out 0.2s;
  :hover {
    background-color: var(--gray-200);
    border-color: var(--gray-400);
  }
`;

const CategoryCard = (props: any) => {
    let navigate = useNavigate();
    let headerFormatted = props.header;
    headerFormatted = capitalizeStr(headerFormatted);
    headerFormatted = replaceDashToSpace(headerFormatted);

    return (
            <StyledCard
                onClick={() => navigate(`${routes.products.path}/1?category=${props.header}`)}
                className="mb-2"
                data-testid="category-card"
            >
                
                <Card.Body className={'d-flex flex-column justify-center'}>
                    
                    <Card.Title className='m-0'>{headerFormatted}</Card.Title>
                    
                    {props.picture && <img alt={'No material'} src={props.picture} className={'img-fluid align-self-center'}
                                           style={{maxHeight: '256px'}}/>}
                </Card.Body>
            </StyledCard>
    );
};

export default CategoryCard;
