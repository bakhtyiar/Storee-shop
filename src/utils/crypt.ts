import {encryptKey} from "./constants";
const CryptoJS = require("crypto-js");

export const encryptText = (text: string, key: string = encryptKey): string => {
    const ciphertext = CryptoJS.AES.encrypt(JSON.stringify(text), key).toString();
    return (ciphertext);
}

export const decryptText = (ciphertext: string, key: string = encryptKey): string => {
    const bytes = CryptoJS.AES.decrypt(ciphertext, key);
    const originalText = bytes.toString(CryptoJS.enc.Utf8);
    return (originalText);
}

export const encryptObject = (data: Object, key: string = encryptKey): string => {
    const ciphertext = CryptoJS.AES.encrypt(JSON.stringify(data), key).toString();
    return (ciphertext);
}

export const decryptObject = (data: string, key: string = encryptKey): Object => {
    const bytes  = CryptoJS.AES.decrypt(data, key);
    const decryptedData = JSON.parse(bytes.toString(CryptoJS.enc.Utf8));
    return (decryptedData);
}