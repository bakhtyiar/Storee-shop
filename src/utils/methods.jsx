import {dummyjsonURL, pageLimit} from './constants';

export const getData = async (skip = 0, limit = pageLimit, url = dummyjsonURL) => {
	return (fetch(`${url}?skip=${skip}&limit=${limit}`) 
	.catch(e => console.log(e))
	.then(response => response.json())
	.then(data => {
		console.log(data);
		return data;
	}))
};