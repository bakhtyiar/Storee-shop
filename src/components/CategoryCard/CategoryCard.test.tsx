import CategoryCard from "./CategoryCard";
import {screen} from "@testing-library/react";
import {renderWithRouter} from "../../../tests/integration-helpers/renderWithRouter";
import {routes} from "../../utils/constants";

describe('CategoryCard test', () => {
    test('Match snapshot', () => {
        renderWithRouter(<CategoryCard header={''}/>, routes.notFound.path);
        expect(screen.getByTestId("category-card")).toMatchSnapshot();
    })
})
