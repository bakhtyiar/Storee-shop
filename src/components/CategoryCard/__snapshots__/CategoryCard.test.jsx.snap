// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard test 1 1`] = `
Array [
  <div
    class="mb-2 card bg-light"
    data-testid="categoryCard"
  >
    <div
      class="d-flex flex-column justify-center card-body"
    >
      <div
        class="m-0 card-title h5"
      >
        Smartphones
      </div>
      <img
        alt="No material"
        class="img-fluid align-self-center"
        src="smartphones.png"
        style="max-height: 256px;"
      />
    </div>
  </div>,
  <div
    class="mb-2 card bg-light"
    data-testid="categoryCard"
  >
    <div
      class="d-flex flex-column justify-center card-body"
    >
      <div
        class="m-0 card-title h5"
      >
        Skincare
      </div>
      <img
        alt="No material"
        class="img-fluid align-self-center"
        src="skincare.png"
        style="max-height: 256px;"
      />
    </div>
  </div>,
  <div
    class="mb-2 card bg-light"
    data-testid="categoryCard"
  >
    <div
      class="d-flex flex-column justify-center card-body"
    >
      <div
        class="m-0 card-title h5"
      >
        Furniture
      </div>
      <img
        alt="No material"
        class="img-fluid align-self-center"
        src="furniture.png"
        style="max-height: 256px;"
      />
    </div>
  </div>,
  <div
    class="mb-2 card bg-light"
    data-testid="categoryCard"
  >
    <div
      class="d-flex flex-column justify-center card-body"
    >
      <div
        class="m-0 card-title h5"
      >
        Sunglasses
      </div>
      <img
        alt="No material"
        class="img-fluid align-self-center"
        src="sunglasses.png"
        style="max-height: 256px;"
      />
    </div>
  </div>,
  <div
    class="mb-2 card bg-light"
    data-testid="categoryCard"
  >
    <div
      class="d-flex flex-column justify-center card-body"
    >
      <div
        class="m-0 card-title h5"
      />
    </div>
  </div>,
]
`;

exports[`CategoryCard test Match snapshot 1`] = `
<div
  class="mb-2 card bg-light"
  data-testid="category-card"
>
  <div
    class="d-flex flex-column justify-center card-body"
  >
    <div
      class="m-0 card-title h5"
    />
  </div>
</div>
`;
